### ISSUE TITLE
PHP 5.3 support

### ISSUE THREAD

**Issue by thunderer** [ehough/shortcode#1]
You may be interested in fact that I changed my mind and "fixed" code for PHP 5.3, current release 0.2.2 contains necessary changes. I appreciate the work you put here but PHP 5.2 it's dead for more than [4 years](http://php.net/eol.php) now and everyone *should* be now working with at least 5.5 (5.4 is only on security fixes). I know that real world projects often lag behind cutting edge software but we should not support this. Your work inspired me though to tinker around the autoloader to automatically rewrite namespaces, maybe it'll be good to go without rewriting code.

Also, if you're not adding anything new to the library I'd be happy if you kept my namespace (Thunder_Shortcode_*) and authorship comment at the beginning of every file.

**Comment by thunderer** [comment]
Also regarding PHP support [WordPress stats](https://wordpress.org/about/stats/) are much more credible and they report only 15% of hosts on PHP 5.2. We should really not support that, for our good as developers and for the whole world to finally move on.

**Comment by thunderer** [comment]
@ehough I've just seen your comment on my blog, if 5.2 is really something you personally need to support then no worries, I understand. But please try to upgrade as soon as possible. :)

**Comment by ehough** [comment]
Thanks for your comments!

> I appreciate the work you put here but PHP 5.2 it's dead for more than 4 years now and everyone should be now working with at least 5.5 (5.4 is only on security fixes). I know that real world projects often lag behind cutting edge software but we should not support this. 

All true. Believe me I hate PHP 5.2, and making software compatible with PHP 5.2 is contributing to the problem as it makes it easier for users to not upgrade. But I'm also stuck with a couple hundred paying customers that are running 5.2.x on their servers and can't afford to drop them completely. Thankfully 5.2 usage is steadily declining over time. Once it hits the single digits I'll probably drop support for it.

> Also, if you're not adding anything new to the library I'd be happy if you kept my namespace (Thunder_Shortcode_*) and authorship comment at the beginning of every file.

No problem. Your authorship comment should be intact already, and I'll change the prefixes today.

**Comment by thunderer** [comment]
@ehough Thanks, I understand you as I moved from such area years ago. Are you actually using this library in production? May I see some links to places that are powered by Shortcode? If you agree I'd be interested in creating a "showcase" (in README for example) to build credibility. I'm running it myself in production but you know, author alone won't make it. :)

Also if anything is troubling you or you have any ideas that would make Shortcode better, I'm all ears. Thanks again for your work and kind cooperation.

**Comment by ehough** [comment]
@thunderer Yes I'm preparing to use the library in the next version of [TubePress Pro](http://tubepress.com/). We need nested shortcode support and our existing homebrewed library doesn't cut it. Will ping you once it's released into production.

I like the idea of adding a showcase! I might also send you a couple PRs as I dig deeper and start putting your code to work.

**Comment by thunderer** [comment]
@ehough Nice, I checked out TubePress and it's a really nice project. Any ETA on the release?

If you have any other ideas (maybe even not related to shortcodes) that can be turned into project or library, I'd be happy to hear them. For example I just released another library called [Numbase](https://github.com/thunderer/Numbase) for number conversion, could be useful for, say, link shorteners to convert IDs to slugs.
